plugins/search-backend/src/service/router.test.ts

```diff
@@ -208,5 +211,55 @@ describe('createRouter', () => {
         });
       });
     });
+
+    describe('Be less restrictive with unknown keys on query endpoint', () => {
+      const expectedResult = {
+        type: 'software-catalog',
+        document: {
+          text: 'foo',
+          title: 'bar baz',
+          location: '/catalog/default/component/example',
+        },
+      };
+      beforeAll(async () => {
+        const logger = getVoidLogger();
+        mockSearchEngine = {
+          getIndexer: jest.fn(),
+          setTranslator: jest.fn(),
+          query: jest.fn(async query => {
+            const obj: IndexableResultSet = {
+              results: [],
+              nextPageCursor: '',
+              previousPageCursor: '',
+            };
+            // @ts-ignore
+            if (query.unknownKey1 && query.unknownKey2) {
+              obj.results.push(expectedResult);
+              return obj;
+            }
+            return obj;
+          }),
+        };
+        const indexBuilder = new IndexBuilder({
+          logger,
+          searchEngine: mockSearchEngine,
+        });
+        const router = await createRouter({
+          engine: indexBuilder.getSearchEngine(),
+          types: indexBuilder.getDocumentTypes(),
+          config: new ConfigReader({ permissions: { enabled: false } }),
+          permissions: mockPermissionEvaluator,
+          logger,
+        });
+        app = express().use(router);
+      });
+      it('pass through unknown keys', async () => {
+        const queryString =
+          'term=test&%5BdocType%5D%5B0%5D=Service&filters%5BdocType%5D%5B0%5D=filter1&unknownKey1%5B2%5D=unknownValue1&unknownKey1%5B3%5D=unknownValue2&unknownKey2=unknownValue1&pageCursor';
+        const response = await request(app).get(`/query?${queryString}`);
+        expect(response.status).toEqual(200);
+        expect(response.body).toMatchObject({ results: [expectedResult] });
```

Comment:
Skimming through this test, it's a little unclear how checking the response body validates that unknown keys can be passed through.  It looks like most of that logic lives in the test setup.

To make it easier for future developers who stumble upon this code, maybe intsead of asserting on the result body, you could directly assert on the argument passed to the query?

Something like..

```ts
expect(mockSearchEngine.query).toHaveBeenCalledWith(expect.objectContaining({
  unknownKey1: [/* the expected value */],
  unknownKey2: '/* the expected value */'
});
```